### PR TITLE
Changes for GEOS-Chem v14.4 within CESM compatibility

### PR DESCRIPTION
**Pull request by jimmielin**
### Name and Institution (Required)

Name: Haipeng Lin
Institution: Harvard Univ.

### Describe the update

These are updates that only affect GEOS-Chem within CESM to make the model compatible with coupling to CESM.

There are several (minor) changes bundled in this update.
* `GeosCore/ucx_mod.F90`: remove `MPI_Bcast` call. This was previously removed in https://github.com/geoschem/geos-chem/pull/1993 but was somehow regressed in the latest version.
* `run/CESM/geoschem_config.yml`: Alphabetical ordering of complex SOA species. The `run/CESM` config file is not generated by the run directory script, but the run directory script is also updated in https://github.com/geoschem/geos-chem/pull/2346 for consistency across versions.
* `run/CESM/HEMCO_Config.rc`: disable ParaNOx extension by default in CESM due to causing issues.

### Expected changes

Affects GEOS-Chem within CESM only.

### Reference(s)

N/A

### Related Github Issue

Companion PR in geoschem/CAM development repository for GEOS-Chem within CESM: https://github.com/geoschem/CAM/pull/34

Tagging @lizziel. Thanks!

### CODE REVIEW

File: GeosCore/ucx_mod.F90 [hunk 1]
```diff
@@ -4139,7 +4139,6 @@ SUBROUTINE NOXCOEFF_INIT( Input_Opt, State_Grid, State_Chm )
 #if defined( MODEL_CESM )
     ENDIF
 
-    CALL MPI_BCAST( State_Chm%NOXCOEFF, nSize, mpi_real8, masterprocid, mpicom, ierr )
```
Comment (lizziel) [inline GeosCore/ucx_mod.F90:4142]: The error catch right below this line should be removed as well.

**Review by lizziel — CHANGES_REQUESTED** [review]

File: GeosCore/ucx_mod.F90 (hunk 1 above)
Comment (jimmielin) [reply GeosCore/ucx_mod.F90:4142]: Thanks Lizzie, I also got rid of all the unnecessary variable definitions.

**Review by lizziel — APPROVED** [review]

**Comment by yantosca** [comment]
Thanks @lizzie and @jimmielin!  I'll pull this into no-diff-to-benchmark.

**Comment by yantosca** [comment]
All GCClassic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #220f810 GEOS-Chem submod update: Merge PR #2346 and PR #2349 (Fixes for CESM)
GEOS-Chem #40a0723cc Merge PR #2348 (Changes for 14.4.0 within CESM compatibility)
HEMCO     #55d27e2 Merge PR #280 (Now use comma-separated list in stale.yml )

Using 24 OpenMP threads
Number of execution tests: 28

Submitted as SLURM job: 37492456
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Furthermore, all tests were zero-diff w/r/t PR #2346 and PR #2349, except:
- TOMAS (parallelization issue?)

**Comment by yantosca** [comment]
All GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

GCHP      #97d30de GEOS-Chem submod update: Merge PR #2346 and PR #2349 (Fixes for CESM)
GEOS-Chem #40a0723cc Merge PR #2348 (Changes for 14.4.0 within CESM compatibility)
HEMCO     #

Number of execution tests: 11

Submitted as SLURM job: 37493005
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
All tests were zero-diff w/r/t the PR #2346 and PR #2349.

**Pull request merged**